### DISCUSSION TITLE
如何自由移动节点的 label

### DISCUSSION THREAD

**Discussion by dukeP** [antvis/X6#3079] [Q&A] [answered]
edge 的 label 可以通过 edgeLabelMovable 配置是否可以移动，node 以及 node-editor 似乎没有相应的配置，请问如何实现此类操作？

**Comment by dukeP** [comment, marked as answer]
已实现：
移动指定 label 需要在注册节点时添加自定义属性来作为标识；可以通过事件系统计算移动位置，利用 refX2、refY2 设置 label 位置；要注意的是需要利用标识在拖动指定 label 时禁止节点移动（interacting.nodeMoveable）
![20221223-150548](https://user-images.githubusercontent.com/17401244/209289727-38d2bd33-8b49-4385-9529-e65c428b3372.gif)

**Comment by dukeP** [comment]
另外，对于一个节点有多个 label （多个 text 节点），并且需要单独编辑的场景也可以采用类似的方式：
label 添加自定义属性记录 selector -> 事件系统中捕获操作 -> 根据 selector 设置对应 label 的 attrs
![20221223-154921](https://user-images.githubusercontent.com/17401244/209295844-744e0e0e-4e11-4d0d-ae48-49f9f8416ef8.gif)

**Reply by SingleDogNo1** [reply]
大哥还在不，小弟也遇到了类似的需求但实在愚钝看不懂，如果看到了可以贴一段代码不，谢谢了

**Reply by dukeP** [reply]
这都两年前的了，不知道api变了不。。。。核心就是定义节点的时候给需要交互的地方加上标记，然后在定义事件的时候根据标记做相应的操作：
![image](https://github.com/user-attachments/assets/df6a1737-ab02-4eca-b508-a16b4f78de6b)
这个节点对 label2 加了个moveable标记
![image](https://github.com/user-attachments/assets/93fdaaf4-44ea-4b59-bbf5-1f537b391b06)
这里就在对应事件里检查标记做操作（更新节点属性）就行了
![image](https://github.com/user-attachments/assets/5eee45a7-66f6-49c6-831e-cf5346cc0344)

**Reply by SingleDogNo1** [reply]
👌🏻，谢谢，我找找文档
